go-controller/pkg/ovn/ovn.go

```diff
@@ -542,15 +543,29 @@ func (oc *Controller) iterateRetryPods(updateAll bool) {
 		backoff := (podEntry.backoffSec * time.Second) + (time.Duration(rand.Intn(500)) * time.Millisecond)
 		podTimer := podEntry.timeStamp.Add(backoff)
 		if updateAll || now.After(podTimer) {
-			klog.Infof("%s retry pod setup", podDesc)
+			if !podEntry.ignoreAddRetry {
```

Comment:
So I think we need to tweak this function or the overall logic a bit. I'm thinking we need to iterate over all of the to be deleted pods first, and then iterate over all of the retry add pods. Then we need to guard deletes from in process adds. Consider this scenario 1 with a pod named "A":
1. pod A fails to deleted, its entry is in the cache and updated so retryCache[uuid1]{retryEntry with pod A}
2. pod A is recreated, initRetry pod adds a new entry to the retryCache[uuid2]{retryEntry with pod A}, pod ensure succeeds because LSP will already exist in DB, we will get that and just update it, uuid2 entry is removed from cache
3. retry loop runs, retryCache uuid1 goes and deletes the LSP just setup in 2

Consider this scenario 2 with a pod named "A":
1. pod A fails to deleted, its entry is in the cache and updated so retryCache[uuid1]{retryEntry with pod A}
2. pod A is recreated, initRetry pod adds a new entry to the retryCache[uuid2]{retryEntry with pod A}, pod ensure fails
3. retry loop runs, retryCache uuid2 runs first because adds are processed first, pod A LSP is updated, next retryCache uuid1 is deleted, and deletes the LSP updated for uuid2

Reply:
Also further compounding this problem is that we use the portCache (a different cache) to get information to use in the delete operation (like IPs) and this cache is keyed based on pod name. That means if an ADD happens for the same name, we will write the new IP information, making a later delete for an old pod UID accidentally delete routes and other things for the new IP!

Reply:
Another option is to store a copy of the portCache entry in the retry map and delete it from the portCache when we are going to retry. You could even use that as your signal something needs to be deleted, rather than having your extra boolean. Then in retry if you see something in that field, you delete that first, then if ignore is set to false, you add the new thing after.